File: singleNumber.py
def single_number (self,nums):
    for i in nums:
        if(len(nums)<= 2):
            print(nums[0])
        else:
            if(nums[i+1] == nums[i]):
                i+1
            elif(nums[i+1] != nums[i]):
                print(nums[i])
            elif(nums[i] == nums[len-1(nums)]):
                print(nums[i])

File: singleNumber.js

var singleNumber = function(nums) {
    nums.sort(function(a, b){return a-b});
    for(i=0;i<nums.length;i++){
        if(nums.length <= 2){
            return nums[0]
        }else{
            if(nums[i+1] == nums[i]){
                i+=1
            }else if(nums[i+1] != nums[i]){
                return nums[i]
            }else if(nums[i] == nums[nums.length-1]){
                return nums[i]
            }
        }
    }
};

console.log(singleNumber([1,2,5,2,5,4,4,8,1]))